drivers/containerd/containerd.go

```diff
@@ -0,0 +1,993 @@
+// +build linux
```

Comment:
is there any code coming from containerd and we need to maintain the copyright header?

Reply:
No.
And before I waste your time...

I don't intend to get graphdriver part (f98ab81) merged into libpod immediately. At this moment, it's just an example implementation and it's incomplete at some points,

- f98ab81 relies on stargz-snapshotter but it will require users to run additional daemons (=containerd+stargz-snapshotter) and it might not fit to the podman's design phirosophy?
  - we might need a daemonless implementation of this patch.
- stargz-snapshotter is untested in the rootless world. (so the patch works only with root user)
- some implementation details are imcomplete yet (e.g. the way to treat id mappings, lockfiles, ...)

The patch f98ab81 intends to make sure that "layer discovery" functionality (56035c0) is needed to make remote-snapshotter-like functionality come true in podman.
I think the deeper review (or design discussion) of the graphdriver implementation can be after the "layer discovery" part (56035c0) is done (But it's kinda chicken and egg problem...).

Please tell me if I should separate the PR.